interface/rest/shops_handler.go

```diff
@@ -21,14 +21,21 @@ func (s *shopsHandler) Search(w http.ResponseWriter, req *http.Request) {
 	// TODO: logging
 	query := req.URL.Query()
 
-	// TODO: validation all inputs
+	// TODO: validation of all inputs
 	search := query.Get("query")
 	lat, _ := strconv.ParseFloat(query.Get("lat"), 32)
 	long, _ := strconv.ParseFloat(query.Get("long"), 32)
 
 	if len(search) == 0 {
-		// TODO: make error object
-		respond(w, http.StatusBadRequest, "must provide a search query")
+		result, err := s.searchShopService.GetShops(req.Context())
```

Comment:
You probably want to limit this somehow, right? Pagination or something, else thousands of things could come back here?